internal/types/action/action.go

```diff
@@ -215,29 +215,61 @@ func (act Type) IsActionOrParent(suspect Type) bool {
 	return strings.HasPrefix(suspect.String(), fmt.Sprintf("%s:", act.String()))
 }
 
-// ActionSet stores a slice of action types
-type ActionSet []Type
+// ActionSet is a set of action.Type.
+type ActionSet map[Type]struct{}
+
+// NewActionSet creates an ActionSet.
+func NewActionSet(t ...Type) ActionSet {
+	a := make(ActionSet, len(t))
+	a.Add(t...)
+	return a
+}
+
+// Add adds the provided Types t to a, any duplicates are ignored.
+func (a ActionSet) Add(t ...Type) {
+	for _, tt := range t {
+		a[tt] = struct{}{}
+	}
+}
+
+// ActionSetUnion returns a new ActionSet that is the union sets.
+func ActionSetUnion(sets ...ActionSet) ActionSet {
+	a := make(ActionSet)
+	for _, s := range sets {
+		for k := range s {
+			a.Add(k)
+		}
+	}
+	return a
+}
+
+// ActionSetDifference returns a new ActionSet that is the difference of a-b.
+func ActionSetDifference(a ActionSet, b ActionSet) ActionSet {
```

Comment:
```suggestion
func Difference(a ActionSet, b ActionSet) ActionSet {
```
Why not just `Difference`?